### PR TITLE
Remove tensorflow dependency

### PR DESCRIPTION
**Pull request by bputzeys**

### CODE REVIEW

File: helical/models/classification/neural_network.py [hunk 1]
```diff
@@ -2,18 +2,25 @@
...
+import torch.nn as nn
+import torch.optim as optim
+from torch.utils.data import DataLoader, TensorDataset
+import logging
 
+LOGGER = logging.getLogger(__name__)
 class NeuralNetwork(BaseTaskModel):
-    def __init__(self, loss: str = "categorical_crossentropy", learning_rate: float = 0.001, epochs=10, batch_size=32) -> None:
-        self.loss = loss
+    def __init__(self, loss: str = "categorical_crossentropy",learning_rate: float = 0.001, epochs=10, batch_size=32) -> None:
```
Comment (giogix2) [inline helical/models/classification/neural_network.py:15]: nit: we could probably rename "categorical_crossentropy" as just "crossentropy". The motivation for this is that categorical_crossentropy is a term used mostly in Keras. In Pytorch the CrossEntropyLoss works almost the same in the same way, but with a different name.

It's a very minor comment. Feel free to ignore.

Comment (mattwoodx) [reply helical/models/classification/neural_network.py:15]: Maybe we can take the loss function in as a parameter instead of a string because this limits us to only the loss functions we have implemented?

File: helical/models/classification/neural_network.py [hunk 2]
```diff
@@ -32,18 +39,20 @@ def compile(self, num_classes: int, input_shape: int) -> None:
...
+        self.input_shape = input_shape
+        self.model = nn.Sequential(
+            nn.Linear(input_shape, 256),
+            nn.ReLU(),
+            nn.Dropout(0.4),
+            nn.Linear(256, 64),
+            nn.ReLU(),
+            nn.Dropout(0.4),
+            nn.Linear(64, num_classes)
+        )
```
Comment (mattwoodx) [inline helical/models/classification/neural_network.py:51]: I know this is how we had it before, but do we want this to be configurable at all?

File: helical/models/classification/neural_network.py [hunk 3]
```diff
@@ -32,18 +39,20 @@ def compile(self, num_classes: int, input_shape: int) -> None:
...
+
+        # Set optimizer and loss function
+        self.optimizer = optim.Adam(self.model.parameters(), lr=self.learning_rate)
```
Comment (mattwoodx) [inline helical/models/classification/neural_network.py:54]: Should we take this in as a param as well?

File: helical/models/classification/neural_network.py [hunk 4]
```diff
@@ -85,9 +133,12 @@ def predict(self, x: ndarray) -> ndarray:
...
         """
-        predictions = self.model.predict(x)
-        y_pred = np.argmax(predictions, axis=1)
-        return self.encoder.inverse_transform(y_pred)
+        self.model.eval()
+        X_tensor = torch.tensor(x, dtype=torch.float32)
+        with torch.no_grad():
+            outputs = self.model(X_tensor)
+            _, predicted = torch.max(outputs, 1)
+        return self.encoder.inverse_transform(predicted.numpy())
```
Comment (mattwoodx) [inline helical/models/classification/neural_network.py:141]: Is this necessary?

Comment (bputzeys) [reply helical/models/classification/neural_network.py:141]: thanks for catching this!

File: helical/models/classification/neural_network.py (hunk 2 above)
Comment (bputzeys) [reply helical/models/classification/neural_network.py:51]: there is a TODO but so far there has been no request to make it configurable

**Review by giogix2 — APPROVED** [review]
